File: 백준/Silver/1874. 스택 수열/스택 수열.py
import sys

n = int(sys.stdin.readline())

stack_list = [int(sys.stdin.readline()) for _ in range(n)]
stack = [1]
result = ['+']
idx = 1
try:
    while len(stack_list) != 0:
        if stack:
            if stack[-1] == stack_list[0]:
                result.append('-')
                stack.pop(-1)
                stack_list.pop(0)
            elif stack[-1] < stack_list[0] or len(stack) == 0:
                for num in range(idx+1,stack_list[0]+1):
                    result.append('+')
                    stack.append(num)
                idx = stack_list[0]
            else:
                raise ValueError()
                break
        else:
            for num in range(idx+1,stack_list[0]+1):
                result.append('+')
                stack.append(num)
            idx = stack_list[0]
    print('\n'.join(result))
except:
    print('NO')

File: node.js/백준/Silver/1874. 스택 수열/스택 수열.js
const fs = require('fs');
const inputs = fs.readFileSync('/dev/stdin').toString().trim().split("\n").map(Number);
const stack = [];
let idx = 1;
let ans = [];
for (let i = 1; i <= inputs[0] ; i++){
    stack.push(i);
    ans.push("+");

    while (stack.length > 0 && inputs[idx] === stack[stack.length-1]){
        stack.pop();
        ans.push("-");
        idx++;
    }
}
if (stack.length >= 1) {
    ans = ["NO"];
}
console.log(ans.join("\n"));